packages/amplify-graphql-api-construct/src/types.ts

```diff
@@ -0,0 +1,222 @@
+import { Duration, CfnResource } from 'aws-cdk-lib';
+import {
+  SchemaFile,
+  CfnGraphQLApi,
+  CfnGraphQLSchema,
+  CfnApiKey,
+  CfnResolver,
+  CfnFunctionConfiguration,
+  CfnDataSource,
+} from 'aws-cdk-lib/aws-appsync';
+import { CfnTable } from 'aws-cdk-lib/aws-dynamodb';
+import { IRole, CfnRole, CfnPolicy } from 'aws-cdk-lib/aws-iam';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { ResolverConfig } from '@aws-amplify/graphql-transformer-core';
+import { TransformParameters, TransformerPluginProvider } from '@aws-amplify/graphql-transformer-interfaces';
+
+export type IAMAuthorizationConfig = {
+  identityPoolId?: string;
+  authRole?: IRole;
+  unauthRole?: IRole;
+  adminRoles?: IRole[];
+};
+
+export type UserPoolAuthorizationConfig = {
+  userPool: IUserPool;
+};
+
+export type OIDCAuthorizationConfig = {
+  oidcProviderName: string;
+  oidcIssuerUrl: string;
+  clientId?: string;
+  tokenExpiryFromAuth: Duration;
+  tokenExpiryFromIssue: Duration;
+};
+
+export type ApiKeyAuthorizationConfig = {
+  description?: string;
+  expires: Duration;
+};
+
+export type LambdaAuthorizationConfig = {
+  function: IFunction;
+  ttl: Duration;
+};
+
+/**
+ * Authorization Config to apply to the API.
+ * At least one config must be provided, and if more than one are provided,
+ * a defaultAuthMode must be specified.
+ * For more information on Amplify API auth, refer to https://docs.amplify.aws/cli/graphql/authorization-rules/#authorization-strategies
+ */
+export type AuthorizationConfig = {
```

Comment:
Should we expose this or allow something like
```
const authConstruct = new AmplifyAuth(....);
const dataConstruct = new GraphQlApi(...);
dataConstruct.authorizeWith(authConstruct);
```
perhaps these are not mutually exclusive.